src/sgame/sg_bot_nav.cpp

```diff
@@ -598,10 +598,25 @@ bool BotMoveToGoal( gentity_t *self )
 	glm::vec3 dir = self->botMind->nav().glm_dir();
 	if ( BotAvoidObstacles( self, dir ) )
 	{
+		// determine a new direction by going a bit sideway
+
+		int clientNum = self - g_entities;
+		dir[2] = 0;
+		dir = glm::normalize(dir);
+		glm::vec3 up = { 0, 0, 1 };
+		glm::vec3 right = glm::cross(dir, up);
+
+		// invert periodically four times per second. We add a delta
+		// determined from clientNum to avoid bots dancing in sync
+		bool invert = ((110 * clientNum + self->client->time1000) % 125) < 250;
```

Comment:
The condition is always true, barring integer overflows. Did you test this?

Reply:
Good catch!

Bots still moved randomly by changing target, that made it look about right :eyes:. Fixed. I think it looks better now.